### PR TITLE
Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by Margen67**
Heavily based off of https://github.com/hax0kartik/luma-hourlies by @hax0kartik.

Example run: https://github.com/Margen67/Luma3DS/runs/554977195

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Build
+      env:
+        ctr_ver: 16
+      run: |
+        docker pull devkitpro/devkitarm
+        docker run -e ENABLE_COMPATIBILITY_REPORTING -v `pwd`:`pwd` devkitpro/devkitarm /bin/bash -c "\
+        . /etc/profile.d/devkit-env.sh && \
+        sudo apt-get update && \
+        sudo apt-get install p7zip-full python3 python3-pip -y && \
+        git clone --depth=1 https://github.com/smealum/ctrulib.git && \
```
Comment (TuxSH) [inline .github/workflows/CI.yml]: I don't think this is needed, is it? Will check if docker image is up to date?

**Review by TuxSH — CHANGES_REQUESTED** [review]
In general, why reinstall everything at every CI trigger? Is it not possible to use a docker image with everything set up instead?

**Comment by Margen67** [comment]
> 
> 
> In general, why reinstall everything at every CI trigger? Is it not possible to use a docker image with everything set up instead?

As far as I know that would require making a Dockerfile/image.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (TuxSH) [reply .github/workflows/CI.yml]: nevermind, relies on acu

Comment (Margen67) [reply .github/workflows/CI.yml]: I tried a Luma3DS Docker image, but it fails:
https://hub.docker.com/r/nanquitas/luma3ds
https://github.com/Margen67/Luma3DS/runs/599002429

If the Docker images were up-to-date that would make things easier: https://github.com/devkitPro/docker/issues/6

**Comment by Margen67** [comment]
Thanks to @Nanquitas updating their Docker image this can be done cleanly now.
@jbmagination you might want to use this improved version.

**Comment by suprstarrd** [comment]
@TuxSH?

Also yes, I will use that version.

**Pull request closed without merging**